enterprise_access/apps/api/v1/tests/test_views.py

```diff
@@ -0,0 +1,369 @@
+        )
+        self.user_license_request_2 = LicenseRequestFactory(
+            enterprise_customer_uuid=self.enterprise_customer_uuid_2,
+            lms_user_id=self.user.lms_user_id
+        )
+
+        # license request under the user's enterprise but not for the user
+        self.enterprise_license_request = LicenseRequestFactory(
+            enterprise_customer_uuid=self.enterprise_customer_uuid_1
+        )
+
+        # license request with no associations to the user
+        self.other_license_request = LicenseRequestFactory()
+
+
+    def test_list_as_enterprise_learner(self):
+        """
+        Test that an enterprise learner should see all their requests.
+        """
+
+        self.set_jwt_cookie(roles_and_contexts=[
+            {
+                'system_wide_role': SYSTEM_ENTERPRISE_LEARNER_ROLE,
+                'context': str(self.enterprise_customer_uuid_1)
+            },
+            {
+                'system_wide_role': SYSTEM_ENTERPRISE_LEARNER_ROLE,
+                'context': str(self.enterprise_customer_uuid_2)
+            }
+        ])
+
+        response = self.client.get(LICENSE_REQUESTS_LIST_ENDPOINT)
+        response_json = self.load_json(response.content)
+        license_request_uuids = sorted([lr['uuid'] for lr in response_json['results']])
+        expected_license_request_uuids = sorted([
+            str(self.user_license_request_1.uuid),
+            str(self.user_license_request_2.uuid)
+        ])
+        assert license_request_uuids == expected_license_request_uuids
+
+    def test_list_as_enterprise_admin(self):
+        """
+        Test that an enterprise admin should see all their requests and requests under their enterprise.
+        """
+
+        self.set_jwt_cookie([{
+            'system_wide_role': SYSTEM_ENTERPRISE_ADMIN_ROLE,
+            'context': str(self.enterprise_customer_uuid_1)
+        }])
+
+        response = self.client.get(LICENSE_REQUESTS_LIST_ENDPOINT)
+        response_json = self.load_json(response.content)
+
+        license_request_uuids = sorted([lr['uuid'] for lr in response_json['results']])
+        expected_license_request_uuids = sorted([
+            str(self.user_license_request_1.uuid),
+            str(self.user_license_request_2.uuid),
+            str(self.enterprise_license_request.uuid)
+        ])
+        assert license_request_uuids == expected_license_request_uuids
+
+    def test_create_no_customer_configuration(self):
+        """
+        Test that a 422 response is returned when creating a request
+        if no customer configuration is set up.
+        """
+        payload = {
+            'enterprise_customer_uuid': self.enterprise_customer_uuid_1,
+            'course_id': 'edx-demo'
+        }
+        response = self.client.post(LICENSE_REQUESTS_LIST_ENDPOINT, payload)
+        assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
+        assert response.data == 'Customer configuration for enterprise: ' + \
```

Comment:
i personally don't love `\` in code in general, but especially for string concatenation. I usually go for ()

like,
```
expected_message = (
  'Customer configuration for enterprise: '
  f'{self.enterprise_customer_uuid_1} does not exist.'
)
assert assert response.data == expected_message
```

just me though 🤷

Reply:
no preference here either :p will update